opal/browser/utils.rb

```diff
@@ -23,6 +34,17 @@ def self.included(klass)
 
   module NativeCachedWrapperClassMethods
     def new(native)
+      # We can't access arbitrary properties if an element is restricted
+      # i.e. the DOM element is an item we can't fully access due to CORS.
+      if Browser::restricted?(native)
+        # Let's try to bypass any further initializers... may be ugly, but
+        # works.
+        obj = allocate
+        obj.instance_variable_set :@native, native
+        return obj
+      end
```

Comment:
About this I would suggest to move all the restricted logic here, and do the check within x-strings, catching just the security error instead of a generic `JS::Error`. This will make things faster, allow unrelated errors to correctly pop up, and keep this specific code inside `NativeCachedWrapperClassMethods` until a more general way of dealing with this kind of problem is found.

Reply:
I think about expanding the scope of that someday in the future.

It would be nice to have something similar to ```Native(`[]`)```. Like ```Browser.native(`iframe.contentWindow`)``` which would return an element of type `Browser::Window`.

This would probably involve using `Object.getPrototypeOf` and it could allow us to drop a warning if types don't match.